File: src-tauri/src/classes/inventory.rs
#[allow(dead_code)]

pub mod inventory {
    use serde::{Deserialize, Serialize};
    #[derive(Debug, Serialize, Deserialize, Clone)]
    pub struct Inventory {
        right_hand: String,
        left_hand: String,
        head: String,
        torso: String,
        legs: String,
        objects: Vec<String>,
    }

    impl Inventory {
        pub fn new() -> Inventory {
            Inventory {
                right_hand: String::from("Sword"),
                left_hand: String::from("Shield"),
                head: String::from("Helmet"),
                torso: String::from("Armor"),
                legs: String::from("Legs"),
                objects: Vec::new(),
            }
        }

        pub fn add_objects(&mut self, items: Vec<String>) {
            for item in items {
                self.objects.push(String::from(item))
            }
        }
    }
}

File: src/models/inventory.js
import Loot from './loot';

export default class Inventory {
    constructor(data) {
        this.gold = data.gold;
        this.objects = data.objects.map((obj) => new Loot(obj));
        this.head = data.head ? new Loot(data.head) : null;
        this.legs = data.legs ? new Loot(data.legs) : null;
        this.torso = data.torso ? new Loot(data.torso) : null;
        this.right_hand = data.right_hand ? new Loot(data.right_hand) : null;
        this.left_hand = data.left_hand ? new Loot(data.left_hand) : null;
    }
}
